### ISSUE TITLE
Add documentation for `--key` and `--cert` parameters

### ISSUE THREAD

**Issue by dmitrizagidulin** [linkeddata/ldnode#173]
Add some more docs to the `--key` and `--cert` parameters to the readme.
Specifically:

- What they're used for
- Whether or not they're required for other parameters (like WebID+TLS)
- Instructions how to generate the key/certificate.

**Comment by nicola** [comment]
+1!

**Comment by dmitrizagidulin** [comment]
Ok, Documentation for the `key`/`cert` parameters and their relationship to other parameters was added in the above PR.

Still needed:

- Add (or link to) instructions on how to generate a simple self-signed certificate, for testing.

**Comment by melvincarvalho** [comment]
By the way --cert and --key are commonly used in curl

       --key <key>
              (SSL/SSH) Private key file name. Allows you to provide your private key in this separate file.

              If this option is used several times, the last one will be used.


       -E, --cert <certificate[:password]>
              (SSL)  Tells  curl  to  use  the specified client certificate file when getting a file with HTTPS, FTPS or another SSL-based protocol. The certificate must be in PKCS#12 format if using
              Secure Transport, or PEM format if using any other engine.  If the optional password isn't specified, it will be queried for on the terminal. Note that this option assumes  a  "certifi‐
              cate" file that is the private key and the private certificate concatenated! See --cert and --key to specify them independently.

              If  curl  is  built against the NSS SSL library then this option can tell curl the nickname of the certificate to use within the NSS database defined by the environment variable SSL_DIR
              (or by default /etc/pki/nssdb). If the NSS PEM PKCS#11 module (libnsspem.so) is available then PEM files may be loaded. If you want to use a file from the current directory, please pre‐
              cede  it with "./" prefix, in order to avoid confusion with a nickname.  If the nickname contains ":", it needs to be preceded by "\" so that it is not recognized as password delimiter.
              If the nickname contains "\", it needs to be escaped as "\\" so that it is not recognized as an escape character.

              (iOS and Mac OS X only) If curl is built against Secure Transport, then the certificate string can either be the name of a certificate/private key in the system or user keychain, or the
              path to a PKCS#12-encoded certificate and private key. If you want to use a file from the current directory, please precede it with "./" prefix, in order to avoid confusion with a nick‐
              name.

              If this option is used several times, the last one will be used.

**Comment by dmitrizagidulin** [comment]
@melvincarvalho - Thanks! I haven't dealt with PKCS#12-encoded certificates that much, can you give an example of how to generate a self-signed one using `openssl`?

**Comment by melvincarvalho** [comment]
Perhaps the easiest way is to export from the browser

**Comment by csarven** [comment]
I've documented that here:

http://www.w3.org/wiki/index.php?title=Foaf%2Bssl/HOWTO#How_to_create_a_certificate_which_includes_the_WebID_URI_for_multi-purpose_use

**Comment by nicola** [comment]
make sure to mention to have CN = '*.yourHost' or there will be pain with your identity provider!

**Comment by nicola** [comment]
This is my tip

```
$ openssl genrsa 2048 > host.key
$ openssl req -new -x509 -nodes -sha1 -days 3650 -key host.key -subj '/CN=*.localhost' > host.cert
```

**Comment by csarven** [comment]
+1 to @nicola 's suggestion. It is simpler and to the point of what's required for this issue. Mine includes other use cases. Same result in the end.

**Comment by dmitrizagidulin** [comment]
@nicola - I can't get your suggestion above to work. What am I doing wrong?

```bash
openssl genrsa 2048 > localhost.key
openssl req -new -x509 -nodes -sha1 -days 3650 -key localhost.key -subj '/CN=*.localhost' > localhost.cert

ldnode --webid --port 443 --cert ../localhost.cert --key ../localhost.key -v
  ldnode:settings root: /Users/dmitriz/code/CrosscloudProj/ldnode/ +0ms
  ldnode:settings URI path filter regexp: undefined +1ms
  ldnode:settings Verbose: true +1ms
  ldnode:settings WebID: true +1ms
  ldnode:settings Live: true +0ms
  ldnode:settings mount: / +5ms
  ldnode:settings SSL Private Key path: ../localhost.key +0ms
  ldnode:settings SSL Certificate path: ../localhost.cert +0ms
  ldnode:settings Certificate: -----BEGIN CERTIFICATE-----
MIIDIDCCAgigAwIBAgIJALGZQNJaGKGqMA0GCSqGSIb3DQEBBQUAMBQxEjAQBgNV
...
-----END CERTIFICATE-----
 +0ms
events.js:85
      throw er; // Unhandled 'error' event
            ^
Error: listen EACCES
    at exports._errnoException (util.js:746:11)
    at Server._listen2 (net.js:1139:19)
    at listen (net.js:1182:10)
    at Server.listen (net.js:1267:5)
    at Object.<anonymous> (/Users/dmitriz/.nvm/versions/node/v0.12.7/lib/node_modules/ldnode/bin/ldnode.js:134:5)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
```

I also tried just `'/CN=localhost'` (instead of `'/CN=*.localhost'` as above):

```
openssl req -new -x509 -nodes -sha1 -days 3650 -key localhost.key -subj '/CN=localhost' > localhost.cert
```

but with same result.

**Comment by nicola** [comment]
`Error: listen EACCES` means that you very likely have this port already open or you don't have enough permissions. (443 is reserved for root!)

**Comment by dmitrizagidulin** [comment]
@nicola - ::facepalms:: yep, that's exactly it! Missed it!!
Thanks :)

**Comment by nicola** [comment]
@dmitrizagidulin we need a better error, you are right.

**Comment by dmitrizagidulin** [comment]
@csarven - thanks for that wiki link, btw, very helpful!

**Comment by dmitrizagidulin** [comment]
Last of the docs merged in #180, closing.
